### PR TITLE
FOGL-7685 python logger statements updated to get multiline traceback if exception occurred

### PR DESCRIPTION
**Pull request by ashish-jabble**

### CODE REVIEW

File: python/fledge/common/microservice_management_client/microservice_management_client.py [hunk 1]
```diff
@@ -86,8 +87,7 @@ def unregister_service(self, microservice_id):
         try:
             response["id"]
         except (KeyError, Exception) as ex:
-            _logger.exception("Could not unregister the micro-service having uuid %s, Reason: %s",
-                              microservice_id, str(ex))
+            _logger.exception(ex, "Could not unregister the micro-service having uuid {}".format(microservice_id))
```
Comment (praveen-garg) [inline python/fledge/common/microservice_management_client/microservice_management_client.py:90]: All good. Though I see we are using microservice, micro-service, service at different places for the same thing. uuid -> UUID

File: python/fledge/common/plugin_discovery.py [hunk 2]
```diff
@@ -204,9 +204,9 @@ def get_plugin_config(cls, plugin_dir, plugin_type, installed_dir_name, is_confi
         except DeprecationWarning:
             _logger.warning('"{}" plugin is deprecated'.format(plugin_dir.split('/')[-1]))
         except FileNotFoundError as ex:
-            _logger.error('Plugin "{}" import problem from path "{}". {}'.format(plugin_dir, plugin_module_path, str(ex)))
+            _logger.error(ex, 'Plugin "{}" import problem from path "{}".'.format(plugin_dir, plugin_module_path))
```
Comment (praveen-garg) [inline python/fledge/common/plugin_discovery.py:207]: Import problem from path {} for plugin {}

File: python/fledge/common/plugin_discovery.py [hunk 3]
```diff
@@ -204,9 +204,9 @@ def get_plugin_config(cls, plugin_dir, plugin_type, installed_dir_name, is_confi
...
         except Exception as ex:
-            _logger.exception('Plugin "{}" raised exception "{}" while fetching config'.format(plugin_dir, str(ex)))
+            _logger.exception(ex, 'Plugin "{}" failed while fetching config'.format(plugin_dir))
```
Comment (praveen-garg) [inline python/fledge/common/plugin_discovery.py:209]: Plugin did not fail, it is -  

Failed to fetch config for plugin {}

File: python/fledge/common/statistics.py [hunk 4]
```diff
@@ -143,7 +143,7 @@ async def register(self, key, description):
             """ The error may be because the key has been created in another process, reload keys """
             await self._load_keys()
             if key not in self._registered_keys:
-                _logger.exception('Unable to create new statistic %s, error %s', key, str(ex))
+                _logger.exception(ex, 'Unable to create new statistic {}'.format(key))
```
Comment (praveen-garg) [inline python/fledge/common/statistics.py:146]: `Unable to create new statistic key {}`?

File: python/fledge/services/common/microservice.py [hunk 5]
```diff
@@ -85,7 +85,7 @@ def __init__(self):
             res = self.register_service_with_core(self._get_service_registration_payload())
             self._microservice_id = res["id"]
         except Exception as ex:
-            _logger.exception('Unable to intialize FledgeMicroservice due to exception %s', str(ex))
+            _logger.exception(ex, 'Unable to initialize FledgeMicroservice')
```
Comment (praveen-garg) [inline python/fledge/services/common/microservice.py:88]: Fledge Service | Fledge Microservice

File: python/fledge/services/common/utils.py [hunk 6]
```diff
@@ -63,7 +63,7 @@ async def shutdown_service(service, loop=None):
                 if not status_code == 200:
                     raise Exception(message=text)
     except Exception as ex:
-        _logger.exception('Error in Service shutdown %s, %s', service._name, str(ex))
+        _logger.exception(ex, 'Error in {} Service shutdown'.format(service._name))
```
Comment (praveen-garg) [inline python/fledge/services/common/utils.py:66]: May be ERROR is redundant. 

Failed to shut down {} service.

File: python/fledge/services/core/api/audit.py [hunk 7]
```diff
@@ -270,7 +270,7 @@ async def get_audit_entries(request):
             res.append(r)
     except Exception as ex:
         msg = str(ex)
-        _logger.error("Get Audit log entry failed. {}".format(msg))
+        _logger.error(ex, "Get Audit log entry failed.")
```
Comment (praveen-garg) [inline python/fledge/services/core/api/audit.py:273]: Failed to get Audit log entry!

File: python/fledge/services/core/api/backup_restore.py [hunk 8]
```diff
@@ -116,7 +116,7 @@ async def get_backups(request):
             res.append(r)
     except Exception as ex:
         msg = str(ex)
-        _logger.error("Get all backups failed. {}".format(msg))
+        _logger.error(ex, "Get all backups failed.")
```
Comment (praveen-garg) [inline python/fledge/services/core/api/backup_restore.py:119]: Failed to get the list of Backup records!

File: python/fledge/services/core/api/backup_restore.py [hunk 9]
```diff
@@ -206,7 +206,7 @@ async def get_backup_download(request):
         raise web.HTTPNotFound(reason=msg, body=json.dumps({"message": msg}))
     except Exception as ex:
         msg = str(ex)
-        _logger.error("Failed to backup download for ID:<{}>. {}".format(backup_id, msg))
+        _logger.error(ex, "Failed to backup download for ID: <{}>.".format(backup_id))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/backup_restore.py:209]: download Backup file

File: python/fledge/services/core/api/common.py [hunk 10]
```diff
@@ -170,7 +170,7 @@ def do_shutdown(request):
         loop = request.loop
         asyncio.ensure_future(server.Server.shutdown(request), loop=loop)
     except RuntimeError as e:
-        _logger.error("Error while stopping Fledge server: {}".format(str(e)))
+        _logger.error(e, "Error while stopping Fledge server.")
```
Comment (praveen-garg) [inline python/fledge/services/core/api/common.py:173]: I think e is string here. (RuntimeError )

File: python/fledge/services/core/api/utils.py [hunk 11]
```diff
@@ -28,16 +28,16 @@ def get_plugin_info(name, dir):
         res = out.decode("utf-8")
         jdoc = json.loads(res)
     except (OSError, ValueError) as err:
-        _logger.error("{} C plugin get info failed due to {}".format(name, str(err)))
+        _logger.error(err, "{} C plugin get info failed.".format(name))
```
Comment (praveen-garg) [inline python/fledge/services/core/api/utils.py:31]: May be Failed to get C based plugin info.

File: python/fledge/services/common/microservice.py (hunk 5 above)
Comment (ashish-jabble) [reply python/fledge/services/common/microservice.py:88]: Class name is `FledgeMicroservice`

File: python/fledge/services/core/api/common.py (hunk 10 above)
Comment (ashish-jabble) [reply python/fledge/services/core/api/common.py:173]: Nope, To get multiline stacktrace we need Exception type.

**Review by praveen-garg — APPROVED** [review]
url -> URL

**Pull request merged**